Weather/ViewModels/CityViewModel.swift

```diff
@@ -0,0 +1,204 @@
+//
+//  CityViewModel.swift
+//  Weather
+//
+//  Created by Alex on 31.07.22.
+//
+
+import Foundation
+import SwiftUI
+import CoreLocation
+
+class CityViewModel: ObservableObject {
+    
+    @Published var weather = WeatherResponse.emptyResponse()
+    @Published var city: String = "Minsk" {
+        didSet {
+            getLocation()
+        }
+    }
+    
+    private lazy var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .full
+        return formatter
+    }()
+    
+    private lazy var dayFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "EEE"
+        return formatter
+    }()
+    
+    private lazy var timeFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "hh a"
+        return formatter
+    }()
+    
+    init() {
+        getLocation()
+    }
+    
+    var date: String {
+        return dateFormatter.string(from: Date(timeIntervalSince1970: TimeInterval(weather.current.dt)))
+    }
+    
+    var weatherIcon: String {
+        if weather.current.weather.count > 0 {
+            return weather.current.weather[0].icon
+        }
+        return "sun.max.fill"
+    }
+    
+    func getTemperature(temp: Double) -> String {
+        return String(format: "%0.1f", temp)
+    }
+    
+    var temperature: String {
+        return getTemperature(temp: weather.current.temp)
+    }
+    
+    var conditions: String {
+        if weather.current.weather.count > 0 {
+            return weather.current.weather[0].main
+        }
+        return ""
+    }
+    
+    var windSpeed: String {
+        return String(format: "%0.1f", weather.current.wind_speed)
+    }
+    
+    var humidity: String {
+        return String(format: "%d%%", weather.current.humidity)
+    }
+    
+    var rainChances: String {
+        return String(format: "%0.0f%%", weather.current.dew_point)
+    }
+    
+    func getTimeFor(timeStamp: Int) -> String {
+        return timeFormatter.string(from: Date(timeIntervalSince1970: TimeInterval(timeStamp)))
+    }
+    
+    func getDayFor(timeStamp: Int) -> String {
+        return dayFormatter.string(from: Date(timeIntervalSince1970: TimeInterval(timeStamp)))
+    }
+    
+    func getWeather(coordinate: CLLocationCoordinate2D?) {
+        if let coordinate = coordinate {
+            let urlString = APIService.getURLFor(lat: coordinate.latitude, lon: coordinate.longitude)
+            getWeatherInternal(city: city, for: urlString)
+        } else {
+            let urlString = APIService.getURLFor(lat: 53.9000, lon: 27.5666)
+            getWeatherInternal(city: city, for: urlString)
+        }
+    }
+    
+    func getWeatherInternal(city: String, for urlString: String) {
+        NetworkManager<WeatherResponse>.fetch(for: URL(string: urlString)!) { (result) in
```

Comment:
Please, don't use force unwrap. You can use `guard let`

Reply:
Resolved.)